src/serac/physics/state/state_manager.hpp

```diff
@@ -130,6 +130,39 @@ class StateManager {
     return std::make_shared<QuadratureData<T>>(elems, qpts_per_elem, initial_state);
   }
 
+  /**
+   * @overload
+   *
+   * @tparam T the type to be created at each quadrature point
+   * @param domain The domain to which this buffer will pertain
+   * @param order The order of the discretization of the displacement and velocity fields
+   * @param dim The spatial dimension of the mesh
+   * @param initial_state the value to be broadcast to each quadrature point
+   * @return shared pointer to quadrature data buffer
+   */
+  template <typename T>
+  static std::shared_ptr<QuadratureData<T>> newQuadratureDataBuffer(const Domain& domain, int order, int dim,
+                                                                    T initial_state)
+  {
+    int Q = order + 1;
+
+    std::array<uint32_t, mfem::Geometry::NUM_GEOMETRIES> elems = geometry_counts(domain);
```

Comment:
There is only two lines different in this vs the `mesh_tag` `newQuadratureDataBuffer` function. Can you reduce the duplication between them?

Reply:
Yes. I removed the `mesh_tag` version. It wasn't being used. The solid mechanics module handles the logic of whether the variables are allocated over the entire mesh or a single `Domain`.